src/sentry/static/sentry/app/components/events/interfaces/debugMeta/debugFileDetails/index.tsx

```diff
@@ -87,18 +87,43 @@ class DebugFileDetails extends AsyncComponent<Props, State> {
       return candidates;
     }
 
-    return candidates.map(candidate => {
-      if (!debugFiles.find(debugFile => debugFile.id === candidate.location)) {
+    // Check for unapplied debug files
+    const unAppliedDebugFiles: Candidates = [];
+
+    for (const debugFileIndex in debugFiles) {
+      const debugFile = debugFiles[debugFileIndex];
+
+      if (candidates.find(candidate => candidate.location === debugFile.id)) {
+        continue;
+      }
```

Comment:
We can optimize this a bit by creating a `Set` of `candidate.location` so that you don't have to traverse `candidates` for every `debugFile`.